Set up and Environment

In [2]:
# Set up and Envirionment
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

Data Loading

In [3]:
df = pd.read_csv("CIS9650_Project 01_healthrate.ratemd.ny.csv")

/var/folders/9g/hd6rk0px3dq9grqrwxn981w40000gn/T/ipykernel_53680/2984594286.py:1: DtypeWarning: Columns (31,32,34,36,65,74,75,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292,293,294,295,296,297,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,337,338,339,340,341,344,345,346,347,348,349,350,351,352,353,354,355,356,357,358,359,360,361,362,363,366,367,368,369,370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,404,405,406,407,410,411,412,413,414,415,416,417,418,419,420,421,422,423,424,425,426,427,42

In [6]:

df.shape
# row, column

(143791, 1611)

In [7]:
df.columns.tolist()

['_id',
 'slug',
 'created',
 'modified',
 'id',
 'facet_url',
 'full_name',
 'full_name_specialty',
 'location.id',
 'location.category',
 'location.created',
 'location.modified',
 'location.longitude',
 'location.latitude',
 'location.slug',
 'location.city.id',
 'location.city.province_name',
 'location.city.province_slug',
 'location.city.country_name',
 'location.city.country_slug',
 'location.city.cover_images.253x83',
 'location.city.cover_images.autoxauto',
 'location.city.modified',
 'location.city.created',
 'location.city.deleted',
 'location.city.slug',
 'location.city.name',
 'location.city.cover_image',
 'location.city.province',
 'location.name',
 'location.address',
 'location.suite',
 'location.postal_code',
 'location.city_id',
 'location.phone_number',
 'location.website',
 'location.image',
 'location.images.autoxauto',
 'location.images.77x77',
 'location.images.100x100',
 'location.images.165x165',
 'location.images.70x70',
 'location.rating.cleanliness',
 'locat

Data Preparation (Cleaning the Data)

The raw data set from the CSV file contained 1,611 columns and 143,791 rows. That in itself is too much to analyze, and also contained fields such as images, phone numbers, etc... Field that would be irrelivant to our analysis, and research question. So this calls for some data cleaning!

* We selected a focused subset of 9 relevant variables that describe each doctor’s name, specialty, city, and core rating metrics.

* We renamed the columns to simpler names. Column names were written such as "location.city.name", "rating.average" so we changed them to "city" and "rating_average"

* Rating fields were stored as mixed text/numeric types so we converted the rating columns to numeric using pd.to_numeric(df_clean[col], errors="coerce")

* We filtered out zero rating doctors to make sure we only analyze doctors who have actual patient reviews in order to produce valid insights

* After filtering, we reset the index for smooth readability

In [10]:
# ---------- DATA CLEANING ----------
# Select usable columns
cols_to_keep = [
    "_id", "full_name", "specialty_name", "location.city.name",
    "rating.average", "rating.helpfulness", "rating.punctuality",
    "rating.staff", "rating.count"
]

df_clean = df[cols_to_keep].copy()

# Rename columns to simpler names
df_clean = df_clean.rename(columns={
    "location.city.name": "city",
    "rating.average": "rating_avg",
    "rating.helpfulness": "rating_help",
    "rating.punctuality": "rating_punctuality",
    "rating.staff": "rating_staff",
    "rating.count": "rating_count"
})

# Convert numeric columns to proper numeric types
numeric_cols = [
    "rating_avg", "rating_help", "rating_punctuality",
    "rating_staff", "rating_count"
]

for col in numeric_cols:
    df_clean[col] = pd.to_numeric(df_clean[col], errors="coerce")

# Drop rows with no rating information
df_clean = df_clean.dropna(subset=["rating_avg"])

# Before having this the rating avg and rating_help was 0's
df_clean = df_clean[df_clean["rating_count"] > 0]
df_clean = df_clean[df_clean["rating_avg"] > 0]

# Reset index
df_clean = df_clean.reset_index(drop=True)

df_clean.head() # this is to see the clean data


_id                full_name  \
0  647e89e399dca372f92a0d7c      Dr. Janan S. Sayyed   
1  647e89e499dca372f92a0d91        Dr. Shirin Peters   
2  647e89e499dca372f92a0e14        Dr. Lev L. Barats   
3  647e89e499dca372f92a0ef0  Dr. Harold N. Bornstein   
4  647e89e499dca372f92a0ef2   Dr. Romelle J. Maloney   

             specialty_name           city  rating_avg  rating_help  \
0              Chiropractor  New York City    4.989221     4.993367   
1  Internist / Geriatrician  New York City    5.000000     5.000000   
2  Internist / Geriatrician   Slingerlands    4.000000     3.800000   
3  Internist / Geriatrician  New York City    4.000000     4.200000   
4      Gynecologist (OBGYN)   New Rochelle    4.000000     3.900000   

   rating_punctuality  rating_staff  rating_count  
0            4.985075       4.99005           603  
1            5.000000       5.00000            48  
2            4.100000       3.80000            10  
3            3.800000       4.20000            10  
4            4.100000       3.80000            10

In [11]:
df.head(50) # this is unclean data

_id  \
0   647e89e399dca372f92a0d7c   
1   647e89e499dca372f92a0d91   
2   647e89e499dca372f92a0db9   
3   647e89e499dca372f92a0dba   
4   647e89e499dca372f92a0dbd   
5   647e89e499dca372f92a0dc3   
6   647e89e499dca372f92a0de4   
7   647e89e499dca372f92a0df1   
8   647e89e499dca372f92a0dfe   
9   647e89e499dca372f92a0e14   
10  647e89e499dca372f92a0e22   
11  647e89e499dca372f92a0e3b   
12  647e89e499dca372f92a0e41   
13  647e89e499dca372f92a0e50   
14  647e89e499dca372f92a0e5a   
15  647e89e499dca372f92a0e6c   
16  647e89e499dca372f92a0e6f   
17  647e89e499dca372f92a0e78   
18  647e89e499dca372f92a0e9b   
19  647e89e499dca372f92a0ea7   
20  647e89e499dca372f92a0ebb   
21  647e89e499dca372f92a0ef0   
22  647e89e499dca372f92a0ef2   
23  647e89e499dca372f92a0ef7   
24  647e89e499dca372f92a0f00   
25  647e89e499dca372f92a0f0b   
26  647e89e499dca372f92a0f1e   
27  647e89e499dca372f92a0f37   
28  647e89e499dca372f92a0f3c   
29  647e89e499dca372f92a0f4a   
30  647e89e499dca372f92a0f4d   
31  647e89e599dca372f92a0f66   
32  647e89e599dca372f92a0f68   
33  647e89e599dca372f92a0f7d   
34  647e89e599dca372f92a0f9b   
35  647e89e599dca372f92a0f9c   
36  647e89e599dca372f92a0f9d   
37  647e89e599dca372f92a0fa3   
38  647e89e599dca372f92a0fa4   
39  647e89e599dca372f92a0fb2   
40  647e89e599dca372f92a0fb4   
41  647e89e599dca372f92a0fca   
42  647e89e599dca372f92a0fd7   
43  647e89e599dca372f92a0fe8   
44  647e89e599dca372f92a0fed   
45  647e89e599dca372f92a1033   
46  647e89e599dca372f92a1034   
47  647e89e599dca372f92a103b   
48  647e89e599dca372f92a1069   
49  647e89e599dca372f92a106a   

                                                 slug  \
0         3183031/Dr-Janan+S.-Sayyed-NEW+YORK-NY.html   
1           3345163/Dr-SHIRIN-PETERS-NEW+YORK-NY.html   
2       3081147/Dr-THOMAS+J.-MULHERN-New+York-NY.html   
3           3081144/Dr-ZINA-RUTKIN-GREAT+NECK-NY.html   
4       3081128/Dr-JOHN+S.-CAVALLARO-Brooklyn-NY.html   
5   3081104/Dr-KEITH+P.-GUTZMANN-Howard+Beach-NY.html   
6            3080954/Dr-ARUNA-MISHRA-New+York-NY.html   
7      3080889/Dr-JONATHAN+D.-PORTER-Syracuse-NY.html   
8           3080843/Dr-ANDRZEJ-RIESS-New+York-NY.html   
9        150786/Dr-Lev+L.-Barats-Slingerlands-NY.html   
10                 3080742/XIAONA-QU-New+York-NY.html   
11           3080638/Dr-ANGELA-BRUCE-New+York-NY.html   
12     3080597/Dr-CHETAN-BHAREL-New+Hyde+Park-NY.html   
13  3080537/Dr-ANNE+MARIE-COMBER-East+Greenbush-NY...   
14     3080482/Dr-JONATHAN-SIEGFRIED-New+York-NY.html   
15          3080410/Dr-HELEN+K.-RALL-NEW+YORK-NY.html   
16  3080400/Dr-NATALIYA+A.-PROKOPENKO-New+York-NY....   
17    2904321/Dr-DOTHULIE+M.-HATCHER-New+York-NY.html   
18         2904217/Dr-LAUREN+M.-POST-New+York-NY.html   
19           2904156/Dr-RONALD-BLAND-New+York-NY.html   
20           2904076/Dr-CHI+K.-VU-Plattsburgh-NY.html   
21     164212/Dr-Harold+N.-Bornstein-New+York-NY.html   
22  138916/Dr-Romelle+J.-Maloney-New+Rochelle-NY.html   
23      2903842/Dr-THADEUS+J.-SCHULZ-New+York-NY.html   
24       2903806/Dr-OLGA-BELOSTOTSKY-NEW+YORK-NY.html   
25        2903754/Dr-JOSEPH+A.-ADLER-New+York-NY.html   
26   2903670/Dr-DAVID+I.-KRATENSTEIN-New+York-NY.html   
27                 2903570/ELI-JIANG-New+York-NY.html   
28       2903549/Dr-MARIE+C.-JACQUES-WESTBURY-NY.html   
29        2903477/Dr-GILBERT-SEIFER-Northport-NY.html   
30     2903464/Dr-Jennifer-SUTHERLAND-Oakdale-NY.html   
31      2903387/Dr-MICHAEL+B.-STOKES-New+York-NY.html   
32         2903377/Dr-KARL+H.-PERZIN-New+York-NY.html   
33       2903292/Dr-Michael+C.-PIAZZA-Kenmore-NY.html   
34             2903151/Dr-SYED-JAFFERY-ELMIRA-NY.html   
35            2903147/Dr-JARED-FUSS-Dix+Hills-NY.html   
36  2903144/Dr-RUXANDRA+C.-BADESCU-Huntington-NY.html   
37       2903121/Dr-ELKA+E.-JACOBSON-New+York-NY.html   
38      2903117/Dr-ARMANDO+M.-ALORAN-New+York-NY.html   
39             2903058/Dr-BLAIR-KOKOTEK-Bronx-NY.html   
40       2903050/Dr-DANIEL+J.-BISONO-New+York-NY.html  